MagickCore/nt-base.c

```diff
@@ -1800,7 +1800,14 @@ MagickPrivate DIR *NTOpenDirectory(const char *path)
     MagickPathExtent);
   if (length == 0)
     return((DIR *) NULL);
+#ifdef __MINGW32__
+  wchar_t WCDirectorySeparator[strlen(DirectorySeparator)+1];
```

Comment:
This indeed incorrect but the solution is actually to remove appending the backslash. This is written in the MSDN documentation:

> This parameter should not be NULL, an invalid string (for example, an empty string or a string that is missing the terminating null character), or end in a trailing backslash (\).

I just pushed a patch to resolve this.